storage/azblob/azblob.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package azblob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/mendersoftware/deployments/model"
+	"github.com/mendersoftware/deployments/storage"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+)
+
+const (
+	headerBlobType             = "x-ms-blob-type"
+	headerMSContentDisposition = "x-ms-blob-content-disposition"
+
+	blobTypeBlock = "BlockBlob"
+)
+
+type client struct {
+	*azblob.ContainerClient
+	fileSuffix  *string
+	contentType *string
+	bufferSize  int
+}
+
+type OpError struct {
+	Op      string
+	Message string
+	Reason  error
+}
+
+func (err OpError) Error() string {
+	errStr := "azblob"
```

Comment:
ok, I do not understand the "azblob" as an error string

Reply:
The error expands to `azblob( err.Op)?(: err.Message)?(: err.Reason.Error())?`.